### PR TITLE
PWN-9232 - app loaders with dependencies

### PR DESCRIPTION
**Pull request by eduardmaximovich**
https://p2pvalidator.atlassian.net/browse/PWN-9232

## Description of Work

- Make AppLoader aware dependencies
- Await only for dependant loaders, otherwise it works in non-blocking mode

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/events/AppLoader.kt [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 
-interface AppLoader {
-    suspend fun onLoad()
-    suspend fun onRefresh(): Unit = Unit
-    suspend fun isEnabled(): Boolean = true
+abstract class AppLoader {
+    private val dependencies: MutableList<AppLoader> = mutableListOf()
+    abstract suspend fun onLoad()
+    open suspend fun onRefresh(): Unit = Unit
+    open suspend fun isEnabled(): Boolean = true
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/events/AppLoader.kt:7]: I would rather make it as abstract to make loaders more explicit - you clearly see conditions when they are loading and you won't forget to use FT

File: app/src/main/java/org/p2p/wallet/home/events/AppLoader.kt [hunk 2]
```diff
@@ -1,7 +1,14 @@
 package org.p2p.wallet.home.events
 
-interface AppLoader {
-    suspend fun onLoad()
-    suspend fun onRefresh(): Unit = Unit
-    suspend fun isEnabled(): Boolean = true
+abstract class AppLoader {
+    private val dependencies: MutableList<AppLoader> = mutableListOf()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/events/AppLoader.kt:4]: Maybe Set? To remove duplicate requests

File: app/src/main/java/org/p2p/wallet/home/events/AppLoader.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/events/AppLoader.kt:7]: Upd: I see, they are unique in the Facade

**Review by Davran331122 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/events/AppLoaderFacade.kt [hunk 3]
```diff
@@ -13,17 +13,70 @@ import org.p2p.core.common.di.AppScope
...
+            val dependencies = loader.getDependencies().mapNotNull { dependency ->
+                appLoaders.find { it.id() == dependency.id() }
+            }
+            dependencyGraph[loader] = dependencies
+        }
+
+        return dependencyGraph
+    }
+
+    private fun AppLoader.id(): String = javaClass.name
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/events/AppLoaderFacade.kt:81]: careful with proguard

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/events/AppLoaderFacade.kt (hunk 3 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/home/events/AppLoaderFacade.kt:81]: I've checked feature build, it seems everything works correctly

**Pull request merged**